src/withField.ts

```diff
@@ -0,0 +1,82 @@
+import { ConnectableElement, Constructor, CustomElement, InputValue } from './types'
+import { getEventTarget, getValueFromEventTarget } from './utils'
+
+import { withValue } from './withValue'
+
+/*
+HoC which returns add "handleChange" and "handleBlur" methods for field
+and setListener for "blur" and "change" events
+also wrap in withValue and provide "value"
+Super need to have "name" or "id" attribute
+
+Works both native and custom event.
+Your custom @change event detail need to have "value" prop
+
+captureBlur is useful if you don't use shadowDOM in your Component or if you use it with slots.
+If you use shadowDOM without slots the event will be bubbling (composed) by default and you don't need to use capture for it.
+
+listenChange is useful if you don't use shadowDOM in your Component
+or if your Component dispatch @change event (with options {bubbles: true, composed: true}) instead of use handleChange method.
+It useless if you use shadowDOM in your Component and not dispatch @change event,
+because js will replace event.target from your inner input to your Component 
+*/
+
+export interface FieldConfig {
+  captureBlur: boolean;
+  listenChange: boolean;
+}
+
+export function withField(config: FieldConfig): ReturnType<typeof withFieldExtended>;
+export function withField(Component: Constructor<ConnectableElement>): ReturnType<ReturnType<typeof withFieldExtended>>;
+export function withField(configOrComponent: FieldConfig | Constructor<ConnectableElement>): ReturnType<typeof withFieldExtended> | ReturnType<ReturnType<typeof withFieldExtended>> {
+  if (typeof configOrComponent === 'object') {
+    return withFieldExtended(configOrComponent)
+  }
+  return withFieldExtended()(configOrComponent)
+}
+
+const withFieldExtended = (config: FieldConfig = {captureBlur: true, listenChange: false}) => <T extends Constructor<CustomElement<{name?: string, id?:string}>>>(Component: T) =>
+    class LiteField extends withValue(Component) {
+
+      setValue = (value: InputValue, validate = false): void => {
+        this._formClass.setValue(this.name || this.id, value, validate);
+      }
+
+      handleChange = (event: Event | CustomEvent): void => {
+       const eventTarget = getEventTarget(event)
+        if (eventTarget) {
+          const value = getValueFromEventTarget(eventTarget)
+          this.setValue(value, true)
+        }
+      }
+
+      setTouched = (name: string): void => {
+        this._formClass.setTouched(name);
```

Comment:
Think it will be better to get name from class, and not from argument: 

`this._formClass.setTouched(this.name || this.id);`

It make execution of setTouched easier from inputs components

Reply:
And maybe it will be useful to have argument "validate" here. As in setValue

Reply:
Done:
- `this._formClass.setTouched(this.name || this.id);`,
- add a validate argument (default to `false` for consistency).

> Also it will make withForm.handleBlur simpler

I fail to see how, could you expand on why/how ?

Reply:
> I fail to see how, could you expand on why/how ?

I make a slip here. I mean `withField.handleBlur, but it not make sense too. Your implementation looks exactly as needed!